test-case-reporting/app.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {PageInfo, TestRun, Travis} from 'test-case-reporting';
+import {TestResultRecord} from './src/test_result_record';
+import {dbConnect} from './src/db';
+import Mustache from 'mustache';
+import bodyParser from 'body-parser';
+import express from 'express';
+import fs from 'fs';
+import statusCodes from 'http-status-codes';
+
+const MAX_PAGE_SIZE = 500;
+
+const app = express();
+const jsonParser = bodyParser.json();
+const db = dbConnect();
+const record = new TestResultRecord(db);
+
+let testRunListTemplate = '';
+function renderTestRunList(testRuns: Array<TestRun>): string {
```

Comment:
This would be a little cleaner imo if we used an IIFE to cache testRunListTemplate, but (a) it wouldn't be lazy anymore and (b) it wouldn't be as readable. Thoughts?

Reply:
Daniel is strongly opposed to filesystem I/O on startup, and I'm inclined to stick with the lazy approach as well. I think this is fine as-is

Reply:
SGTM